core/lib/src/request/from_request.rs

```diff
@@ -501,3 +503,56 @@ impl<'a, 'r, T: FromRequest<'a, 'r> + 'a> FromRequest<'a, 'r> for Option<T> {
         }).boxed()
     }
 }
+
+#[cfg(feature = "tls")]
+fn parse_general_name(gn: &x509_parser::extensions::GeneralName<'_>) -> Option<Vec<u8>> {
+    use x509_parser::extensions::GeneralName;
+    match gn {
+        GeneralName::DNSName(dns) => Some(dns.as_bytes().to_vec()),
+        GeneralName::IPAddress(ip) => Some(ip.to_vec()),
+        GeneralName::RFC822Name(email) => Some(email.as_bytes().to_vec()),
+        // TODO
+        _ => None,
+    }
+}
+
+#[cfg(feature = "tls")]
+#[crate::async_trait]
+impl<'a, 'r> FromRequest<'a, 'r> for MutualTlsInfo {
+    type Error = ();
+
+    async fn from_request(request: &'a Request<'r>) -> Outcome<Self, Self::Error> {
```

Comment:
This looks relatively expensive to do in `from_request`, which might be called multiple times on the way to one route handler or will only have one or two of these fields used. Can we move the parsing code (and the `x509-parser` dependency) onto methods of the `MutualTlsInfo` struct in `rocket_http`, and construct `MutualTlsInfo` from only the underlying certificate bytes?